oioioi/base/processors.py

```diff
@@ -34,7 +34,7 @@ def mathjax_location(request):
 
 
 def gravatar(request):
-    if request.user.is_authenticated:
+    if request.user.is_authenticated and request.user.email:
 
         def generator():
             return str(get_gravatar_url(request.user.email, size=25)) or ''
```

Comment:
> a) send mock email if null -> get expected result (ugly) code

I was about to suggest this:
```suggestion
            return str(get_gravatar_url(request.user.email or '', size=25))
```

but it seems that Gravatar then responds with 404 (purely out of spite? Only happens with .jpg suffix which django-gravatar seems to add. Like, it's fine with an all-zeros hash, but not with hash of empty string). So I guess you have to have *some* default still.

Anyway since Gravatar only receives hashes you should be able to use a non-email-like string such as "oioioi"? I don't really see that as ugly, and certainly it sounds more predictable than using an email.

(Please verify what I'm saying before proceeding, cause I don't have a dev env set up, I'm just pasting chunks of Python into REPL and looking to see what happens.)

Reply:
Yes, you are right. I checked it with 'oioioi' and Gravatar produces expected image as a result. Thanks for the help! 
Pushed and awaites @twalen's review.